core/resharding.go

```diff
@@ -132,22 +133,47 @@ func GetEpochFromBlockNumber(blockNumber uint64) uint64 {
 }
 
 // GetShardingStateFromBlockChain will retrieve random seed and shard map from beacon chain for given a epoch
-func GetShardingStateFromBlockChain(bc *BlockChain, epoch uint64) *ShardingState {
+func GetShardingStateFromBlockChain(bc *BlockChain, epoch uint64,
+) (*ShardingState, error) {
 	number := GetBlockNumberFromEpoch(epoch)
-	shardState := bc.GetShardStateByNumber(number)
+	shardState, err := bc.GetShardStateByNumber(number)
```

Comment:
BTW, we have both `ShardingState` and `ShardState` (confusing naming).  The former is a more comprehensive collection of sharding information including randomness, and the latter is just typedef-ed as a list of `Committee`-s.  Maybe we should rename `ShardState` to `Committees` to better reflect this.  Cc: @chaosma